packages/main/src/CalendarLegend.ts

```diff
@@ -0,0 +1,203 @@
+ *
+ * @constructor
+ * @extends UI5Element
+ * @public
+ * @since 1.22.0
+ */
+@customElement({
+	tag: "ui5-calendar-legend",
+	renderer: litRender,
+	styles: CalendarLegendCss,
+	template: CalendarLegendTemplate,
+	dependencies: [CalendarLegendItem],
+})
+@event<CalendarLegendItemSelectionChangeEventDetail>("_calendar-legend-selection-change", {
+	detail: {
+		item: { type: CalendarLegendItem },
+	},
+})
+class CalendarLegend extends UI5Element {
+	/**
+	 * Hides the Today item in the legend.
+	 *
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	hideToday!: boolean;
+
+	/**
+	 * Hides the Selected day item in the legend.
+	 *
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	hideSelectedDay!: boolean;
+
+	/**
+	 * Hides the Non-Working day item in the legend.
+	 *
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	hideNonWorkingDay!: boolean;
+
+	/**
+	 * Hides the Working day item in the legend.
+	 *
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	hideWorkingDay!: boolean;
+
+	/**
+	 * Defines the items of the component.
+	 * @slot items
+	 * @public
+	 */
+	@slot({
+		type: HTMLElement,
+		invalidateOnChildChange: true,
+		individualSlots: true,
+		"default": true,
+	 })
+	items!: Array<CalendarLegendItem>;
+
+	_itemNavigation: ItemNavigation;
+
+	constructor() {
+		super();
+
+		this._itemNavigation = new ItemNavigation(this, {
+			navigationMode: NavigationMode.Horizontal,
+			behavior: ItemNavigationBehavior.Static,
+			getItemsCallback: () => this.focusableElements,
+		});
+	}
+
+	_onMouseDown(e: MouseEvent) {
+		e.stopPropagation();
+		const target = e.target as CalendarLegendItem;
+
+		this._itemNavigation.setCurrentItem(target);
+		this._itemNavigation._focusCurrentItem();
+	}
+
+	_onFocusIn(e: MouseEvent) {
+		const target = e.target as CalendarLegendItem;
+
+		this.fireEvent<CalendarLegendItemSelectionChangeEventDetail>("_calendar-legend-selection-change", {
+			item: target,
+		});
+	}
+
+	_onItemKeyDown(e: KeyboardEvent) {
+		const items = this.focusableElements;
+		const itemsCount = items.length;
+		const currentItem = e.target as CalendarLegendItem;
+		const currentIndex = items.indexOf(currentItem);
+
+		if (isDown(e)) {
+			e.preventDefault();
+			const nextIndex = currentIndex + 1;
+
+			if (nextIndex < itemsCount) {
+				this._itemNavigation.setCurrentItem(items[nextIndex]);
+				this._itemNavigation._focusCurrentItem();
+			}
+		}
+
+		if (isUp(e)) {
+			e.preventDefault();
+			const nextIndex = currentIndex - 1;
+
+			if (nextIndex >= 0) {
+				this._itemNavigation.setCurrentItem(items[nextIndex]);
+				this._itemNavigation._focusCurrentItem();
+			}
+		}
+	}
+
+	get focusableElements() {
+		let allFocusableItems = [...this.shadowRoot!.querySelectorAll<CalendarLegendItem>("[ui5-calendar-legend-item]"), ...this.legendItems];
+		const rearrangedItems: Array<CalendarLegendItem> = [];
+
```

Comment:
I don't like this solution and how this slot works because users have to follow speciifc order when they slot elements in order to make it to looks correct. Please check `column-width` property. This property will make the login in this method redundant.